scapy/packet.py

```diff
@@ -100,7 +101,7 @@ class Packet(six.with_metaclass(Packet_metaclass,  # type: ignore
         "wirelen",
     ]
     name = None
-    fields_desc = []  # type: List[Field[Any, Any]]
+    fields_desc = []  # type: Sequence[Field[Any, Any]]
```

Comment:
Hi, @gpotter2. In a previous version of this series I was able to set the type of `fields_desc` to `Sequence[Field_metaclass]`, and that allowed mypy to accept `ConditionalField` and `MultipleTypeField` instances. For some reason after the rebase this stopped working. I'm not sure how to solve this now. Do you have any suggestion?

Reply:
Hum, this is actually a bit broken. Let me check

Reply:
This should be fixed with https://github.com/secdev/scapy/pull/2936 (simply make `ConditionalField`, ... inherit from `Field[Any, Any]`)

Reply:
I think I tried making `ConditionalField` and `MultipleTypeField` inherit Field at some point but it caused the trick with `__getattr__` to stop working, and instead the superclass's method were being called.

Reply:
You're absolutely right, I'll have to think this through

Reply:
I was thinking that using a Protocol for structural typing might work, though I'm not sure how mypy would handle __getattr__ when matching a type to a protocol.

Reply:
Sorry for the delay. I think we're going to simply add an `AnyField` type. See https://github.com/secdev/scapy/pull/2954

Reply:
Looks good. I'll try to rebase on top of that and see.